src/method/query/response.rs

```diff
@@ -0,0 +1,231 @@
+use std::ops::Deref;
+use std::ops::DerefMut;
+use std::ops::Index;
+use std::slice::SliceIndex;
+
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+use crate::Result;
+use crate::Value;
+
+/// A wrapper type around the list of results for the queries that were returned
+/// by the database.
+///
+/// Provides utility functions to access the result of one specific query, or if
+/// needed, all queries at once.
+///
+#[derive(Debug, Clone)]
+pub struct QueryResponse(Vec<QueryResult>);
+
+impl QueryResponse {
+    /// Constructs an empty [`QueryResponse`]
+    pub fn new() -> Self {
+        Self(Default::default())
+    }
+
+    /// Returns a reference the result for the `n`-th query from the response. If
+    /// no result is found at this index then [None] is returned.
+    pub fn get_query(&self, n: usize) -> Option<&QueryResult> {
```

Comment:
I feel like naming this `get_query` might cause some confusion down the line. `query_result` maybe? @tobiemh Any ideas?

Reply:
Naming ideas for the functions are very welcome, i spent some time looking for intuitive names but it stills feels confusing in some places. Replacing it `query_result` sounds like a good idea to me too, I'll wait for Tobie's suggestion since you pinged him, but if he agrees as well i'll change it to `query_result`

Reply:
Yeah. Naming is hard.

> i spent some time looking for intuitive names

It shows. You did an amazing job. This PR is well thought out.

Reply:
Since Tobie is probably too busy for a naming question and since `query_result` sounds like a good idea i decided to apply the suggestion right away.